### PR TITLE
Add elasticsearch module with stats metricset

### PR DESCRIPTION
**Pull request by ruflin**
This adds the stats metricset for the elasticsearch module. The stats metricset gives and overview over the current state of the cluster.

### CODE REVIEW

File: metricbeat/module/elasticsearch/stats/_meta/data.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+                "memory": {
+                    "bytes": 6647
+                }
+            },
+            "shards": {
+                "failed": 0,
+                "successful": 11,
+                "total": 22
+            },
+            "store": {
```
Comment (joegallo) [inline metricbeat/module/elasticsearch/stats/_meta/data.json]: Is this `store.size.bytes` the total disk space used by indices? If so, that's awesome, if not then I'm not sure we want it.

File: metricbeat/module/elasticsearch/stats/_meta/data.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+                "count": 4,
+                "memory": {
+                    "bytes": 6647
+                }
+            },
+            "shards": {
+                "failed": 0,
+                "successful": 11,
+                "total": 22
+            },
```
Comment (joegallo) [inline metricbeat/module/elasticsearch/stats/_meta/data.json]: I think docs, segments, shards all look fine. I'd like to have an indices section here as well, though, with a total count (or maybe closed and total)? Whatever you think makes sense on that.

File: metricbeat/module/elasticsearch/stats/_meta/data.json (hunk 1 above)
Comment (ruflin) [inline metricbeat/module/elasticsearch/stats/_meta/data.json]: I will add it in a follow up PR.

File: metricbeat/module/elasticsearch/stats/_meta/data.json (hunk 2 above)
Comment (ruflin) [inline metricbeat/module/elasticsearch/stats/_meta/data.json]: will add it in a follow up PR.

**Comment by ruflin** [comment]
jenkins, retest it

File: metricbeat/module/elasticsearch/_meta/docs.asciidoc [hunk 3]
```diff
@@ -0,0 +1,4 @@
+== elasticsearch Module
+
+This is the elasticsearch Module. The elasticsearch module contains a minimal set of metrics to enable monitoring of elasticsearch across multiple versions.
```
Comment (tsg) [inline metricbeat/module/elasticsearch/_meta/docs.asciidoc]: Should we link to xpack monitoring here, as a more complete alternative? Can come in a different PR.

**Comment by tsg** [comment]
Changelog?

File: metricbeat/module/elasticsearch/_meta/docs.asciidoc (hunk 3 above)
Comment (ruflin) [inline metricbeat/module/elasticsearch/_meta/docs.asciidoc]: +1. Will also add it in Kibana.

File: metricbeat/module/elasticsearch/stats/_meta/data.json (hunk 1 above)
Comment (ruflin) [inline metricbeat/module/elasticsearch/stats/_meta/data.json]: @joegallo The es docs state: `The size of the index.` So I think it is what we assumed above.

File: metricbeat/module/elasticsearch/stats/_meta/data.json (hunk 2 above)
Comment (ruflin) [inline metricbeat/module/elasticsearch/stats/_meta/data.json]: added in https://github.com/elastic/beats/pull/4077

**Pull request merged**